apps/storybook-e2e/src/e2e/components/rich-data-table/vl-rich-data-table.stories.cy.ts

```diff
@@ -1,14 +1,267 @@
+
+const shouldMatchCellWithData = (row: JQuery<HTMLElement>, rowIndex: number, rowData: unknown[]) => {
+    cy.wrap(row)
+        .find('td')
+        .each((cell, cellIndex) => {
+            const dataRowObject = rowData[rowIndex];
+            const dataRowValuesList = Object.values(dataRowObject);
+            const valueForCell = dataRowValuesList[cellIndex];
+            // in the data used, complex data objects are used for custom template instead of simply showing the value
+            // in that case, it won't match one on one
+            if (typeof valueForCell !== 'object') {
+                cy.wrap(cell).should('have.text', dataRowValuesList[cellIndex]);
+            }
+        });
+};
+
+const shouldNotMatchCellWithData = (row: JQuery<HTMLElement>, rowIndex: number, rowData: unknown[]) => {
+    cy.wrap(row)
+        .find('td')
+        .each((cell, cellIndex) => {
+            const dataRowObject = rowData[rowIndex];
+            if (dataRowObject) {
+                const dataRowValuesList = Object.values(dataRowObject);
+                const valueForCell = dataRowValuesList[cellIndex];
+                // in the data used, complex data objects are used for custom template instead of simply showing the value
+                // in that case, it won't match one on one
+                if (typeof valueForCell !== 'object') {
+                    cy.wrap(cell).should('not.have.text', dataRowValuesList[cellIndex]);
+                }
+            }
+        });
+};
+
+const shouldMatchRowCountWithDataLength = (rowData: unknown[]) => {
+    cy.get('vl-rich-data-table').shadow().find('tbody').find('tr').its('length').should('eq', rowData.length);
+};
+
+const shouldMatchTableData = (rowData: unknown[]) => {
+    shouldMatchRowCountWithDataLength(rowData);
+
+    executeForEveryRow((row, rowIndex) => shouldMatchCellWithData(row, rowIndex, rowData));
+};
+
+const getTitleForField = (field) => {
+    switch (field) {
+        case 'name':
+            return 'Naam';
+        case 'id':
+            return 'ID';
+    }
+};
+const shouldSortFieldsAsExpected = (
+    row: JQuery<HTMLElement>,
+    rowIndex: number,
+    dataRows: unknown[],
+    sortField: string,
+    direction = 'asc'
+) => {
+    const listForField = (rows: unknown[]) => rows.map((row) => row[sortField]);
+    const unsortedValues = listForField(dataRows);
+    const valuesAfterSorting = direction === 'asc' ? unsortedValues.sort() : unsortedValues.sort().reverse();
+    cy.wrap(row)
+        .find(`[data-title="${getTitleForField(sortField)}"]`)
+        .should('have.text', valuesAfterSorting[rowIndex]);
+};
+
+const shouldHaveActiveSorterAndMatchExpectedDirection = (sortField: string, direction: string) => {
+    cy.get('vl-rich-data-table')
+        .shadow()
+        .find(`[data-vl-for="${sortField}"]`)
+        .shadow()
+        .find('#direction')
+        .invoke('attr', 'data-vl-icon')
+        .should('eq', direction === 'asc' ? 'arrow-down' : 'arrow-up');
```

Comment:
Deze invoke kan je vervangen door:
`.should('have.attr', 'data-vl-icon', direction === 'asc' ? 'arrow-down' : 'arrow-up');`

Reply:
done